docs/details/shards/General/Clear.md

```diff
@@ -0,0 +1,15 @@
+`Clear` removes all the elements of the sequence that has been passed to it in the `:Name` parameter.
+
+This shard works only on sequences. Parameters `:Key` and `:Global` are not applicable to `Clear`. 
```

Comment:
`:Global`should work here.

`:Key`also if the value associated to that key is a sequence it will clear it. It doesn't clear the table though.

Reply:
`:Global`

I tried, but passing `:Global true` prevents `Clear` from clearing the sequence.

```clj
[1 2 3] >= .seq
(Clear .seq :Global true)
.seq (Log) ;; => [1, 2, 3] 
```

And passing `:Global false` works as expected (i.e. just like without passing `:Global`) because the default value of `:Global` is `false` - it clears the sequence
```clj
[1 2 3] >= .seq
(Clear .seq :Global false)
.seq (Log) ;; => []  
```
```clj
[1 2 3] >= .seq
(Clear .seq )
.seq (Log) ;; => [] 
```

Should I document this behaviour?

Reply:
`:Key`

I checked, `Clear` doesn't seem to work on tables at all (and hence it cannot parse `:Key` parameter either)
```clj
{:k1 1 :k2 2} >= .tbl
(Clear .tbl :Key :k1 )
.tbl (Log) ;; => {}

;; => [trace] [2022-07-27 17:16:50.873] [T-24168] [SHCore.cpp::350] Created a Mesh
;; => Error: Invalid variable
```

Reply:
`:Key`

I checked, `Clear` doesn't work at all on tables (doesn't have any effect), irrespective of the value of the key's value

```clj
{:k1 1 :k2 [11 22]} >= .tbl
.tbl (Log "before") ;; => {k2: [11, 22], k1: 1}
(Clear .tbl :Key "k1" )
.tbl (Log "after")  ;; => {k2: [11, 22], k1: 1}
```
```clj
{:k1 1 :k2 [11 22]} >= .tbl
.tbl (Log "before") ;; => {k2: [11, 22], k1: 1}
(Clear .tbl :Key "k2" )
.tbl (Log "after")  ;; => {k2: [11, 22], k1: 1}
```

Reply:
I think in the generated docs [`(Clear)`](https://docs.fragcolor.xyz/shards/General/Clear/), [`(Count)`](https://docs.fragcolor.xyz/shards/General/Count/) etc are inheriting their function definition, and parameter help text from some base types where these parameters were valid (like `(Set)`, `(Update)`, `(Ref)`, `(Table)`, `(Sequence)`, etc) but here they don't make sense.

I dunno, just a guess.

They all seem to have the same function definition and parameter help text

Reply:
Looks like you are right. Looking at the code I was expecting the behavior I described but it doesn't work like that. Wondering if that is a bug. @sinkingsugar should we maybe rewrite those shards? I'm not a fan of having parameters that either do nothing or not what is expected.

Reply:
Usage of `:Global` documented in commit [#0ecc5a77767aad041cd24a60afb1f15f83bc2a39](https://github.com/fragcolor-xyz/shards/pull/305/commits/0ecc5a77767aad041cd24a60afb1f15f83bc2a39)

Usage of `:Key` shall be documented in issue #335